vulnerabilities/importers/safety_db.py

```diff
@@ -0,0 +1,170 @@
+# Data generated with VulnerableCode require an acknowledgment.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# When you publish or redistribute any data created with VulnerableCode or any VulnerableCode
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+#
+# Data Imported from https://github.com/pyupio/safety-db
+#
+import dataclasses
+import json
+from typing import Any
+from typing import Iterable
+from typing import Mapping
+from typing import Set
+from typing import Tuple
+from urllib.error import HTTPError
+from urllib.request import urlopen
+
+from dephell_specifier import RangeSpecifier
+from packageurl import PackageURL
+from schema import Or
+from schema import Regex
+from schema import Schema
+
+from vulnerabilities.data_source import Advisory
+from vulnerabilities.data_source import DataSource
+from vulnerabilities.data_source import DataSourceConfiguration
+
+
+def validate_schema(advisory_dict):
+
+    scheme = {
+        str:
+            [
+                {
+                    "advisory": str,
+                    "cve": Or(None, Regex(r"CVE-\d+-\d+")),
+                    "id": Regex(r"^pyup.io-\d"),
+                    "specs": list,
+                    "v": str
+                }
+            ]
+    }
+
+    Schema(scheme).validate(advisory_dict)
+
+
+class VersionAPI:
+    def __init__(self, cache=None):
+        self.cache = cache or {}
+
+    def get(self, package_name):
+        if package_name not in self.cache:
+            url = "https://pypi.org/pypi/{}/json".format(package_name)
+            releases = set()
+            try:
+                with urlopen(url) as response:
+                    json_file = json.load(response)
+                    releases = set(json_file['releases'])
+            except HTTPError:  # PyPi does not have data about this package
+                pass
+
+            self.cache[package_name] = releases
+
+        return self.cache[package_name]
+
+
+@dataclasses.dataclass
+class DebianConfiguration(DataSourceConfiguration):
```

Comment:
This seems rather out of place and unused. WIP ?

Reply:
yep :)